app/src/main/java/org/novinger/jason/demoreader/datamodels/Article.java

```diff
@@ -1,79 +1,65 @@
 package org.novinger.jason.demoreader.datamodels;
 
+import com.google.gson.annotations.SerializedName;
+
 import java.util.ArrayList;
 
 /**
- * Created by jason on 4/27/16.
+ * POJO for deserializing articles from the Dailydot feed with gson.
  */
 public class Article {
-    private String id;
-    private int realId;
-    private String headline;
-    private String slug;
-    private ArrayList<Author> authors;
-    private String summary;
-    private String url;
-    private LeadArt lead_art;
-
-    public Article() {}
-
-    public Article(String id, String headline, String slug, ArrayList<Author> authors, String url, String summary, LeadArt lead_art) {
-        this.id = id;
-        this.realId = mungeId(id);
-        this.headline = headline;
-        this.slug = slug;
-        this.authors = authors;
-        this.url = url;
-        this.summary = summary;
-        this.lead_art = lead_art;
-    }
 
-    private int mungeId(String id) {
-        String[] parts = id.split(".");
-        return Integer.parseInt(parts[2]);
-    }
+    @SerializedName("id")
+    private String mId;
 
-    public String toString() {
-        return getHeadline() + " — " + getPrimaryAuthor().getName();
-    }
+    @SerializedName("headline")
+    private String mHeadline;
 
-    public int getId() {
-        return realId;
-    }
+    @SerializedName("slug")
+    private String mSlug;
 
-    public String getHeadline() {
-        return headline;
+    @SerializedName("authors")
+    private ArrayList<Author> mAuthors;
+
+    @SerializedName("summary")
+    private String mSummary;
+
+    @SerializedName("url")
+    private String mUrl;
+
+    @SerializedName("lead_art")
+    private LeadArt mLeadArt;
+
+    public String getId() {
+        return mId;
     }
 
-    public String getSlug() {
-        return slug;
+    public String getHeadline() {
+        return mHeadline;
     }
 
     public ArrayList<Author> getAuthors() {
-        return authors;
+        return mAuthors;
     }
 
     public Author getPrimaryAuthor() {
-        if (authors.size() > 0) {
-            return authors.get(0);
+        if (null != mAuthors && mAuthors.size() > 0) {
```

Comment:
This is an old habit that came from getting NPE when putting the variable first. I don't remember why, but it was beaten in to me at NISC. I remember you telling me it's not any issue anymore (or maybe never was).